contrib x265 fails to build on FreeBSD

### Problem Description

In the master branch, x265 has been updated to x265-snapshot-20220209-12707.
It fails to build with `use of undeclared identifier 'S_IRUSR'` error.
Because `S_IRUSR` is defined in sys/stat.h in FreeBSD, but it's not included.

I'm going to make a pull request for this issue.

### Activity Log, Crash Log or any other details

```text
The error message is shown as follows.


  : /usr/home/yuichiro/HandBrake/build/contrib/x265/x265-snapshot-20220209-12707
/source/common/ringmem.cpp:163:27: error: use of undeclared identifier 'S_IRUSR'

  :             mode_t mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH |
S_IWOTH;
  :                           ^
```
```


### What Operating System are you running?

FreeBSD 14.0-CURRENT

### What version of HandBrake are you running?

head of master branch

### Where did you download HandBrake from?

github

--- a/contrib/x265/P00-freebsd-common-threading.patch
+++ b/contrib/x265/P00-freebsd-common-threading.patch
@@ -1,0 +1,12 @@
+--- a/source/common/threading.h	2022-02-16 10:18:38.000000000 +0900
++++ b/source/common/threading.h	2022-03-02 11:03:42.002417000 +0900
+@@ -37,6 +37,9 @@
+ #include <semaphore.h>
+ #include <errno.h>
+ #include <fcntl.h>
++#ifdef __FreeBSD__
++#include <sys/stat.h>
++#endif
+ #endif
+ 
+ #if MACOS

